src/scripts/create_lt_emotions.py

```diff
@@ -0,0 +1,175 @@
+    test_df1 = dataset["lt_go_emotions_test"].to_pandas()
+    test_df2 = dataset["lt_twitter_emotions_test"].to_pandas()
+
+    # Combine the two test splits
+    test_df = pd.concat([test_df1, test_df2], ignore_index=True)
+
+    def process_split(df: pd.DataFrame) -> pd.DataFrame:
+        """Process a split by filtering single labels and mapping them.
+
+        Args:
+            df: The dataframe to process.
+
+        Returns:
+            The processed dataframe.
+        """
+        # Filter to only keep samples with single labels (ignore multi-label samples)
+        df = df[df["labels"].apply(lambda x: len(x) == 1)].copy()
+
+        # Extract the single integer from the labels list
+        df["label_int"] = df["labels"].apply(lambda x: x[0])
+
+        # Map the labels to the EuroEval sentiment labels
+        label_mapping = {
+            0: "negative",  # anger
+            1: "negative",  # disgust
+            2: "negative",  # fear
+            3: "positive",  # joy
+            4: "negative",  # sadness
+            5: "positive",  # surprise
+            6: "neutral",  # neutral
+        }
+        df["label"] = df["label_int"].map(label_mapping)
+
+        # Remove any unmapped labels
+        df = df.dropna(subset=["label"])
+
+        # Keep only the required columns: lt_text -> text, label
+        df = df[["lt_text", "label"]].rename(columns={"lt_text": "text"})
+
+        # Filter by text length
+        df["text_len"] = df.text.str.len()
+        df = df.query("text_len >= @MIN_NUM_CHARS_IN_DOCUMENT").query(
+            "text_len <= @MAX_NUM_CHARS_IN_DOCUMENT"
+        )
+
+        # Drop the temporary text_len column
+        df = df.drop(columns=["text_len"])
+
+        return df
+
+    # Process each split
+    train_df = process_split(df=train_df)
+    val_df = process_split(df=val_df)
+    test_df = process_split(df=test_df)
+
+    train_size = 1024
+    val_size = 256
+    test_size = 2048
+
+    train_df = create_balanced_split(df=train_df, target_total_size=train_size)
+    val_df = create_balanced_split(df=val_df, target_total_size=val_size)
+    test_df = create_balanced_split(df=test_df, target_total_size=test_size)
+
+    # Reset indices
+    train_df = train_df.reset_index(drop=True)
+    val_df = val_df.reset_index(drop=True)
+    test_df = test_df.reset_index(drop=True)
+
+    # Create DatasetDict
+    dataset = DatasetDict(
+        train=Dataset.from_pandas(train_df, split=Split.TRAIN),
+        val=Dataset.from_pandas(val_df, split=Split.VALIDATION),
+        test=Dataset.from_pandas(test_df, split=Split.TEST),
+    )
+
+    # Create dataset ID
+    dataset_id = "EuroEval/lithuanian-emotions-mini"
+
+    # Remove the dataset from Hugging Face Hub if it already exists
+    HfApi().delete_repo(dataset_id, repo_type="dataset", missing_ok=True)
+
+    # Push the dataset to the Hugging Face Hub
+    dataset.push_to_hub(dataset_id, private=True)
+
+
+def create_balanced_split(
```

Comment:
You could do this a bit easier with scikit-learn's `train_test_split` with the `stratify` argument 🙂